language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -127,6 +139,113 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
         existing_borrows.is_empty()
     }
 
+    fn verify_instantiation(
+        env: (&[StructHandle], &[Kind], usize),
```

Comment:
this should be `self`

Reply:
I actually made it non-self because I thought I would also need it to check struct definitions, which means the "env" must be more generic so it can also work for structs. You mentioned you have already implemented that part so I'll check again.